core/src/main/java/uk/gov/justice/services/core/dispatcher/Dispatcher.java

```diff
@@ -25,7 +38,28 @@ public Dispatcher(final HandlerRegistry handlerRegistry) {
      * @return the envelope returned by the handler method
      */
     public JsonEnvelope dispatch(final JsonEnvelope envelope) {
-        return (JsonEnvelope) handlerRegistry.get(envelope.metadata().name()).execute(envelope);
+
+        final HandlerMethod handlerMethod = handlerRegistry.get(envelope.metadata().name());
+        final Class<?> envelopeGenericType = handlerMethod.getEnvelopeGenericType();
+
+        if(envelopeGenericType == JsonEnvelope.class) {
```

Comment:
I suggest you move the conversion out into a private method so we can just have a single return